specs/BlockchainLayer/EpochManager/RewardsCalculator.md

```diff
@@ -0,0 +1,71 @@
+## Calculating rewards 
+
+Validator rewards are assigned once per epoch.
+- Total reward for the epoch is at most the sum of gas and storage rent paid.
+- A percentage of the reward goes to the treasury account.
+- The reward is split in proportion to stake and adjusted for each validator depending on
+its number of missed blocks and chunks.
+- Extra inflation happens in some way (TODO)
+
+### Parameters
+Rewards calculator uses the following parameters:
+```
+   pub max_inflation_rate: u8,
+   pub num_blocks_per_year: u64,
+   pub epoch_length: u64,
+   pub validator_reward_percentage: u8,
+   pub protocol_reward_percentage: u8,
+   pub protocol_treasury_account: AccountId,
+```
+
+The following values for the epoch:
+```
+    total_storage_rent: Balance,
+    total_validator_reward: Balance,
+    total_supply: Balance,
+```
+
+And values per each validator:
+```
+    stake
+    blocks_expected
+    blocks_produced
+    chunks_expected
+    chunks_produced
+```
+
+### Inflation
+
+```
+max_inflation = (max_inflation_rate * total_supply * epoch_length) / (100 * num_blocks_per_year)
```

Comment:
should unify with economics.
also this is not inflation, but issuance as it can be negative